pkg/services/ngalert/schedule/schedule.go

```diff
@@ -518,15 +528,20 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 						attribute.String("rule_fingerprint", fpStr),
 						attribute.String("tick", utcTick),
 					))
-					defer span.End()
 
-					evaluate(tracingCtx, f, attempt, ctx, span)
-					return nil
-				})
-				if err != nil {
-					logger.Error("Evaluation failed after all retries", "error", err)
+					noRetry := attempt == sch.maxAttempts
```

Comment:
Will this always be false, given `for attempt = 0; attempt < sch.maxAttempts; attempt++ {`?

Reply:
Great catch - I went back and forth on this multiple times, must have done too much ctrl + z 😅